### PR TITLE
Implement `remoteAddress` for `@effect/platform-bun`

### PR DESCRIPTION
**Pull request by Zamion101**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

* Implements `HttpServerRequest.remoteAddress` for `@effect/platform-bun` using `BunServer.requestIp(source)` function.
* Adds Security Warning for use of `xForwardedHeaders`.

### CODE REVIEW

File: packages/platform-bun/src/internal/httpServer.ts [hunk 1]
```diff
@@ -250,7 +250,11 @@ class ServerRequestImpl extends Inspectable.Class implements ServerRequest.HttpS
     return this.source.url
   }
   get remoteAddress(): Option.Option<string> {
-    return this.remoteAddressOverride ? Option.some(this.remoteAddressOverride) : Option.none()
+    return this.remoteAddressOverride
+      ? Option.some(this.remoteAddressOverride)
+      : this.bunServer.requestIP(this.source)
+      ? Option.some(this.bunServer.requestIP(this.source)!.address)
+      : Option.none()
```
Comment (tim-smart) [inline packages/platform-bun/src/internal/httpServer.ts:257]:
```suggestion
      : Option.fromNullable(this.bunServer.requestIP(this.source)?.address)
```

Bit cleaner

File: packages/effect/test/Either.test.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ describe("Either", () => {
       const y = yield* Either.right(2)
       return x + y
     })
-    // eslint-disable-next-line require-yield
+
```
Comment (tim-smart) [inline packages/effect/test/Either.test.ts:25]: If you can remove the unrelated changes :)

**Review by tim-smart — CHANGES_REQUESTED** [review]

File: packages/effect/test/Either.test.ts (hunk 2 above)
Comment (Zamion101) [reply packages/effect/test/Either.test.ts:25]: Sorry `pnpm lint-fix` added automatically. Reverted back.

**Review by tim-smart — APPROVED** [review]
